parser-typechecker/src/Unison/Typechecker/Context.hs

```diff
@@ -1212,13 +1251,69 @@ synthesizeWanted e
       let outputType = existential' l B.Blank outputTypev
       appendContext [existential outputTypev]
       cwant <- checkCases scrutineeType outputType cases
+      ensurePatternCoverage e scrutinee scrutineeType cases
       want <- coalesceWanted cwant swant
```

Comment:
Kind of a basic thing, but I like that you kept the basic structure of the typechecker and just call out to this other module in this one place.